### ISSUE TITLE
[BUG] Pytest Downloads datadownload marked tests

### ISSUE THREAD

**Issue by jgyasu** [sktime/sktime#7607]
**Describe the bug**

I tried running the tests on my local sktime-dev environment and I ran `pytest`, it downloads a directory named `UnitTest/m5-forecasting-accuracy` containing a few `csv` files.

**Expected behavior**

`datadownload` test should be excluded by default when running `pytest`

**Versions**

pytest-8.3.4, pluggy-1.5.0 

System:
    python: 3.12.8 | packaged by Anaconda, Inc. | (main, Dec 11 2024, 16:31:09) [GCC 11.2.0]
executable: /home/jigyasu/anaconda3/envs/sktime-dev/bin/python3
   machine: Linux-6.8.0-50-generic-x86_64-with-glibc2.35

Python dependencies:
          pip: 24.2
       sktime: 0.35.0
      sklearn: 1.5.2
       skbase: 0.12.0
        numpy: 1.26.4
        scipy: 1.14.1
       pandas: 2.2.3
   matplotlib: 3.10.0
       joblib: 1.4.2
        numba: 0.60.0
  statsmodels: 0.14.4
     pmdarima: 2.0.4
statsforecast: 2.0.0
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None

</details>

**Comment by yarnabrina** [comment]
Can you please elaborate why is it a bug, and why these tests should not run by default?

For local development, it's unlikely that you need to run all test cases, very likely you are running only for a particular estimator or a single test for all estimators or at most all tests for a particular capability, e.g. forecasting. Will the following not be enough for that purpose?

```sh
pytest <path to directory containing tests or a specific test module> -k <estimator or test name>
```

**Comment by jgyasu** [comment]
> Can you please elaborate why is it a bug, and why these tests should not run by default?
> 
> For local development, it's unlikely that you need to run all test cases, very likely you are running only for a particular estimator or a single test for all estimators or at most all tests for a particular capability, e.g. forecasting. Will the following not be enough for that purpose?
> 
> ```shell
> pytest <path to directory containing tests or a specific test module> -k <estimator or test name>
> ```

Hmm, that's true. It's unlikely that I will ever need to run all the test cases. I was testing my my development setup and decided to run all the tests and found some dataset being downloaded. I opened this issue after a conversation with @fkiraly in this [thread](https://discord.com/channels/1075852648688930887/1325128732712636466)
